### DISCUSSION TITLE
How to get variable type?

### DISCUSSION THREAD

**Discussion by mateusz-glowinski** [gijzelaerr/python-snap7#437] [General]
Is there a way to get declared variable type?

Let's say I've declared `PLC Tag` `%MB0` named `SomeString` as type `Char`.
Is there a way to check how this memory was marked as?
I want to properly display value after read.

And the same question related to DBs.

Thanks for help in advance.

**Comment by mateusz-glowinski** [comment]
Is it even possible? Or PLC returns `bytearray` and user can interpret it whatever it wants?

**Comment by luewh** [comment]
Hello.

From my knowledge.

`Is there a way to get declared variable?`

PLC only keep the value for the variable, it doesn't keep the name and the data type of the variable, it's TIA portal who does that.

So if you don't know the structure of the data you get from PLC, then you can not decode them (you can guess from which byte to which is an integer...).

But with OPCUA you can load the data structure such as user defined data type etc.

**Reply by mateusz-glowinski** [reply]
Ok, thanks for a tip :)

**Comment by gijzelaerr** [comment]
Closing as answered. PLC memory does not store variable types or names -- only raw byte values. Type information exists only in the TIA Portal project.
